### PR TITLE
[receiver/zookeeper] Add support for ruok command

### LINKED ISSUE #21481: Support ruok command in zookeeper receiver
### Component(s)

receiver/zookeeper

### Is your feature request related to a problem? Please describe.

Other zookeeper agents (e.g., datadog) support the ruok command along with the metric set from mntr, I wanted to inquire regarding adding a scraper to get the ruok response, and potentially adding it as an attribute as it likely doesn't belong in a metric table as a standalone.

### Describe the solution you'd like

We would like the ruok response to be scraped along with the mntr metric set.

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by akats7**
**Description:** <Describe what has changed.>
Add zookeeper ruok health check metric based on response from ruok 4lw command
**NOTE**: preliminary PR not ready to be merged

**Link to tracking Issue:** resolves #21481 

**Testing:**: Modified existing unit tests to account for additional metric from additional command. Made minor change to test structure to account for separate text responses from separate commands. Added unit test to handle following cases, valid "imok" return from server, invalid response from server, and empty response from server.

### CODE REVIEW

File: receiver/zookeeperreceiver/scraper.go [hunk 1]
```diff
@@ -19,12 +19,12 @@ import (
 	"context"
 	"errors"
 	"fmt"
+	"go.opentelemetry.io/collector/pdata/pcommon"
```
Comment (djaglowski) [inline receiver/zookeeperreceiver/scraper.go:22]: This will cause a lint failure. You'll have to move it back

**Review by djaglowski — COMMENTED** [review]
I think with the addition of another connection/call, this scraper's separation of concerns is a bit difficult to follow. Specifically, the way we manage connections and make requests is very mixed together with the way we process responses.

I've opened #22752 to split these concerns apart. Sorry to jump in front of your PR like this but I'll be a lot more confident in the change if you're willing to rebase onto that and follow the pattern there (or let me know why it doesn't work).

**Comment by akats7** [comment]
> I think with the addition of another connection/call, this scraper's separation of concerns is a bit difficult to follow. Specifically, the way we manage connections and make requests is very mixed together with the way we process responses.
> 
> I've opened #22752 to split these concerns apart. Sorry to jump in front of your PR like this but I'll be a lot more confident in the change if you're willing to rebase onto that and follow the pattern there (or let me know why it doesn't work).

Hey @djaglowski, yep that makes perfect sense to me. That does make it a lot cleaner.

**Comment by djaglowski** [comment]
@akats, #22752 was finally merged.

**Comment by akats7** [comment]
> @akats, #22752 was finally merged.

@djaglowski Great thanks, I will push up an update

**Comment by akats7** [comment]
Hey @djaglowski, apologies for the delay, I pushed up an update with the changes

File: receiver/zookeeperreceiver/metadata.yaml [hunk 2]
```diff
@@ -145,3 +145,9 @@ metrics:
       value_type: int
       monotonic: true
       aggregation: cumulative
+  zookeeper.ruok:
+    enabled: true
+    description: Response from zookeeper ruok command
+    unit: 1
+    gauge:
+      value_type: int
```
Comment (djaglowski) [inline receiver/zookeeperreceiver/metadata.yaml:153]:
```suggestion
      value_type: int

```

File: receiver/zookeeperreceiver/testdata/scraper/null-ruok.yaml [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+            sum:
+              aggregationTemporality: 2
+              dataPoints:
+                - asInt: "4"
+                  startTimeUnixNano: "1642685966444471000"
+                  timeUnixNano: "1642685966444739000"
+            unit: '{znodes}'
+        scope:
+          name: otelcol/zookeeperreceiver
+          version: latest
```
Comment (djaglowski) [inline receiver/zookeeperreceiver/testdata/scraper/null-ruok.yaml:148]:
```suggestion
          version: latest

```

**Comment by djaglowski** [comment]
Thanks @akats7. CI is running now. Otherwise, looks good, just a couple minor things.

File: receiver/zookeeperreceiver/testdata/ruok-invalid [hunk 4]
```diff
@@ -0,0 +1 @@
+invalid ruok
```
[the remark this answers is not in the document]
Comment (akats7) [reply receiver/zookeeperreceiver/testdata/ruok-invalid:1]: Hey @djaglowski, no it's just meant to represent any value that's returned other than the expected or null

**Review by djaglowski — APPROVED** [review]

**Comment by djaglowski** [comment]
Thank you @akats7!

**Pull request merged**